components/script/dom/gpucommandencoder.rs

```diff
@@ -106,9 +112,14 @@ impl GPUCommandEncoderMethods for GPUCommandEncoder {
                 // TODO(zakorgy): We should use `_descriptor` here after it's not empty
                 // and the underlying wgpu-core struct is serializable
             ))
-            .expect("Failed to send CopyBufferToBuffer");
+            .expect("Failed to send Finish");
 
         let buffer = receiver.recv().unwrap();
-        GPUCommandBuffer::new(&self.global(), self.channel.clone(), buffer)
+        GPUCommandBuffer::new(
+            &self.global(),
+            self.channel.clone(),
+            buffer,
+            self.buffers.borrow_mut().drain().collect(),
```

Comment:
Let's do this instead: `self.buffers.borrow_mut().drain().map(|b| DomRoot::from_ref(&**b)).collect()`

Reply:
To pass as `DomRoot` here, I had to implement the `Eq` and `Hash` for `DomRoot<GPUBuffer>`.
I've also changed the buffers type in the `GPUCommandEncoder` from Dom<GPUBuffer> to `DomRoot<GPUBuffer>` to be able to use in the `finish` function without using the `unrooted_must_root` annotation.